test/tests/functional/pbs_array_job_mail.py

```diff
@@ -0,0 +1,85 @@
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+
+
+class Test_array_job_email(TestFunctional):
+    """
+    This test suite is for testing arrayjob e-mailing (parent job and subjob)
+    """
+
+    def test_emails(self):
+        """
+        Run arrayjob with -m jabe and test if the e-mails are received
+        """
+
+        mailfile = "/var/mail/%s" % str(TEST_USER)
+        if not os.path.isfile(mailfile):
+            self.logger.info("Mail file '%s' does not exist or "
+                             "mail is not setup. "
+                             "Hence this step would be skipped. "
+                             "Please check manually." % mailfile)
+            return 1
+
+        J = Job(TEST_USER, attrs={ATTR_m: 'jabe', ATTR_J: '1-2'})
+        J.set_sleep_time(1)
+        parent_jid = self.server.submit(J)
+        self.server.expect(JOB, {'job_state': 'B'},
+                           parent_jid, max_attempts=15, interval=2)
+
+        self.logger.info("Wait 10s for array job to finish")
+        time.sleep(10)
+
+        subjob_jid = parent_jid.replace("[]", "[1]")
+
+        emails = [("PBS Job Id: " + parent_jid, "Begun execution"),
+                  ("PBS Job Id: " + parent_jid, "Execution terminated"),
+                  ("PBS Job Id: " + subjob_jid, "Begun execution"),
+                  ("PBS Job Id: " + subjob_jid, "Execution terminated")]
+
+        f = open(mailfile, 'r')
+        maillog = [x.strip() for x in f.readlines()[-600:]]
+
+        for (jobid, msg) in emails:
+            emailpass = 0
+            for i in range(0, len(maillog)-2):
+                if jobid == maillog[i] and msg == maillog[i+2]:
+                    emailpass = 1
+
+            self.assertTrue(emailpass, "Message '" + jobid + " " + msg +
+                            "' not found in " + mailfile)
```

Comment:
Please add one more test with ATTR_m set to "j" and confirm you receive an error message.

Reply:
New test added